unified_planning/model/metrics.py

```diff
@@ -39,27 +46,69 @@ class MinimizeActionCosts(PlanQualityMetric):
 
     def __init__(
         self,
-        costs: Dict["up.model.Action", Optional["up.model.FNode"]],
-        default: Optional["up.model.FNode"] = None,
+        costs: Dict["up.model.Action", Optional["up.model.Expression"]],
```

Comment:
I would remove `Optional` here

```suggestion
        costs: Dict["up.model.Action", "up.model.Expression"],
```